examples/ex18.cpp

```diff
@@ -0,0 +1,304 @@
+//                                MFEM Example 18
+//
+// Compile with: make ex18
+//
+// Sample runs:
+//
+//       ex18 -p 1 -r 2 -o 1 -s 3
+//       ex18 -p 1 -r 1 -o 3 -s 4
+//       ex18 -p 1 -r 0 -o 5 -s 6
+//       ex18 -p 2 -r 1 -o 1 -s 3
+//       ex18 -p 2 -r 0 -o 3 -s 3
+//
+// Description:  This example code solves the compressible Euler system of
+//               equations, a model nonlinear hyperbolic PDE, with a
+//               discontinuous Galerkin (DG) formulation.
+//
+//               Specifically, it solves for an exact solution of the equations
+//               whereby a vortex is transported by a uniform flow. Since all
+//               boundaries are periodic here, the method's accuracy can be
+//               assessed by measuring the difference between the solution and
+//               the initial condition at a later time when the vortex returns
+//               to its initial location.
+//
+//               Note that as the order of the spatial discretization increases,
+//               the timestep must become smaller. This example currently uses a
+//               simple estimate derived by Cockburn and Shu for the 1D RKDG
+//               method. An additional factor can be tuned by passing the --cfl
+//               (or -c shorter) flag.
+//
+//               The example demonstrates user-defined bilinear and nonlinear
```

Comment:
Consider extending/revising this paragraph (also in ex18p)

Reply:
Done in d767dc2.